src/TableFunctions/TableFunctionMerge.cpp

```diff
@@ -191,13 +191,15 @@ ColumnsDescription TableFunctionMerge::getActualTableStructure(ContextPtr contex
 
 StoragePtr TableFunctionMerge::executeImpl(const ASTPtr & /*ast_function*/, ContextPtr context, const std::string & table_name, ColumnsDescription /*cached_columns*/, bool /*is_insert_query*/) const
 {
+    std::optional<std::string> table_to_write = std::nullopt;
```

Comment:
Do we have specific reasons to not enable writes via table function merge?

Reply:
Merge table can merge only part of columns and has no knowledge about rest columns, in this case right way is to write in source table only, I think.
Or I did not understand question?

Reply:
@ianton-ru , Table function Merge https://clickhouse.com/docs/sql-reference/table-functions/merge is a kind of a wrapper around a table with Engine Merge.
So my question is why writes are not possible using Table function Merge. 
I do not mean that writes must be enabled, I am just wondering.

Reply:
Actually writes to Table function Merge seem to be totally pointless.

Reply:
Ah. Yes, I did not add it to table function because this looks useless for function. But for consistency may be need to add too,